CONTRIBUTING.md

```diff
@@ -47,19 +52,56 @@ Linux beefy.machine 4.4.0-101-generic #124-Ubuntu SMP Fri Nov 10 18:29:59 UTC 20
 My system has IPv6 disabled.
 ```
 
-## Writing a Patch
+### Submitting Code Contributions
+
+#### The Development Workflow
+
+1.  **Fork & Clone**: Fork the repository and clone it to your local machine.
+
+    ```bash
+    git clone https://github.com/YOUR_USERNAME/swift-prometheus.git
+    cd swift-prometheus
+    ```
+
+2.  **Create a Branch**: Create a descriptive branch for your changes (e.g., `fix/counter-overflow` or `feature/new-exporter`).
+
+    ```bash
+    git checkout -b fix/counter-overflow
+    ```
+
+3.  **Write Code & Tests**: Make your changes. All new code must be accompanied by tests to prevent regressions. For changes that are performance-critical or security-sensitive, we also strongly encourage adding new benchmarks or fuzz tests.
+
+4.  **Run Checks Locally**: Before pushing, validate your changes by running the project's automated checks on your local machine.
+    - **Run Unit Tests**: This is the most fundamental check to ensure your changes haven't broken existing functionality.
+
+        ```bash
+        swift test
+        ```
+
+    - **Check Code Formatting**: We use `swift-format` to maintain a consistent code style. Run the following command to verify your code is formatted correctly.
+
+        ```bash
+        curl -s --retry 3 https://raw.githubusercontent.com/swiftlang/github-workflows/refs/heads/main/.github/workflows/scripts/check-swift-format.sh | bash
+        ```
+
+    - **Simulate CI Workflows (Optional)**: For a more comprehensive check, you can run the full GitHub Actions workflows locally using [act](https://github.com/nektos/act). This is useful for catching issues that only appear in the CI environment.
+        - For detailed setup instructions, see the Swift project's guide on [Running Workflows Locally](https://github.com/swiftlang/github-workflows?tab=readme-ov-file#running-workflows-locally).
```

Comment:
I tried the whole sequence with `act` and really struggled to get it working with anything OTHER than docker-desktop - I love the concept of describing this and providing a means to check the workflows locally, but it was particularly cranky for me.

I'm not immediately clear on what other options there might be, or how to make that process work a bit more smoothly - say for example, using `container` instead of docker and what might need to exist to arrange that. That's probably beyond the scope of this edit though.

Reply:
> That's probably beyond the scope of this edit though.

I would say so, yes.

Should we open a new issue against https://github.com/swiftlang/github-workflows?tab=readme-ov-file#running-workflows-locally just to keep track of the callouts?
(We should consider proposing various cross-platform solutions.)